py-polars/polars/utils/udfs.py

```diff
@@ -29,72 +30,66 @@ class StackValue(NamedTuple):
 ApplyTarget: TypeAlias = Literal["expr", "frame", "series"]
 StackEntry: TypeAlias = Union[str, StackValue]
 
+_MIN_PY311 = sys.version_info >= (3, 11)
+
+
+class OpNames:
+    BINARY = {
+        "BINARY_ADD": "+",
+        "BINARY_AND": "&",
+        "BINARY_FLOOR_DIVIDE": "//",
+        "BINARY_MODULO": "%",
+        "BINARY_MULTIPLY": "*",
+        "BINARY_OR": "|",
+        "BINARY_POWER": "**",
+        "BINARY_SUBTRACT": "-",
+        "BINARY_TRUE_DIVIDE": "/",
+        "BINARY_XOR": "^",
+    }
+    CALL = "CALL" if _MIN_PY311 else "CALL_*"
+    CONTROL_FLOW = (
+        {
+            "POP_JUMP_FORWARD_IF_FALSE": "&",
+            "POP_JUMP_FORWARD_IF_TRUE": "|",
+            "JUMP_IF_FALSE_OR_POP": "&",
+            "JUMP_IF_TRUE_OR_POP": "|",
+        }
+        if _MIN_PY311
+        else {
+            "POP_JUMP_IF_FALSE": "&",
+            "POP_JUMP_IF_TRUE": "|",
+            "JUMP_IF_FALSE_OR_POP": "&",
+            "JUMP_IF_TRUE_OR_POP": "|",
+        }
+    )
+    LOAD_VALUES = frozenset(("LOAD_CONST", "LOAD_DEREF", "LOAD_FAST", "LOAD_GLOBAL"))
+    LOAD = LOAD_VALUES | {"LOAD_METHOD", "LOAD_ATTR"}
+    SYNTHETIC = {
+        "POLARS_EXPRESSION": 1,
+    }
+    UNARY = {
+        "UNARY_NEGATIVE": "-",
+        "UNARY_POSITIVE": "+",
+        "UNARY_NOT": "~",
+    }
+    PARSEABLE_OPS = (
+        {"BINARY_OP", "COMPARE_OP", "CONTAINS_OP", "IS_OP"}
+        | set(UNARY)
+        | set(CONTROL_FLOW)
+        | set(SYNTHETIC)
+        | LOAD_VALUES
+    )
+    UNARY_VALUES = frozenset(UNARY.values())
 
-# Note: in 3.11 individual binary opcodes were folded into a new BINARY_OP
-_BINARY_OPCODES = {
-    "BINARY_ADD": "+",
-    "BINARY_AND": "&",
-    "BINARY_FLOOR_DIVIDE": "//",
-    "BINARY_MODULO": "%",
-    "BINARY_MULTIPLY": "*",
-    "BINARY_OR": "|",
-    "BINARY_POWER": "**",
-    "BINARY_SUBTRACT": "-",
-    "BINARY_TRUE_DIVIDE": "/",
-    "BINARY_XOR": "^",
-}
-_CONTROL_FLOW_OPCODES = {
-    # note: once we add additional JUMP op support, we'll need to disambiguate
-    # between and/or (what we currently support) and if/else (which we don't)
-    # "POP_JUMP_FORWARD_IF_FALSE": "?",
-    # "POP_JUMP_FORWARD_IF_TRUE": "?",
-    # "POP_JUMP_IF_FALSE": "?",
-    # "POP_JUMP_IF_TRUE": "?",
-    # "JUMP_FORWARD": "?",
-    "JUMP_IF_FALSE_OR_POP": "&",
-    "JUMP_IF_TRUE_OR_POP": "|",
-}
-_UNARY_OPCODES = {
-    "UNARY_NEGATIVE": "-",
-    "UNARY_POSITIVE": "+",
-    "UNARY_NOT": "~",
-}
-_SYNTHETIC_OPS = {
-    "POLARS_EXPRESSION": 1,
-}
-_LOAD_OPS = {
-    "LOAD_CONST",
-    "LOAD_DEREF",
-    "LOAD_FAST",
-    "LOAD_GLOBAL",
-}
-_SIMPLE_EXPR_OPS = {
-    "BINARY_OP",
-    "COMPARE_OP",
-    "CONTAINS_OP",
-    "IS_OP",
-}
-_SIMPLE_EXPR_OPS |= (
-    set(_UNARY_OPCODES) | set(_CONTROL_FLOW_OPCODES) | set(_SYNTHETIC_OPS) | _LOAD_OPS
-)
-_SIMPLE_FRAME_OPS = _SIMPLE_EXPR_OPS | {"BINARY_SUBSCR"}
```

Comment:
getting rid of this til dataframe.apply is supported? if so, +1, makes sense to me

Reply:
> getting rid of this til dataframe.apply is supported? if so, +1, makes sense to me

Yup 👍